### PR TITLE
Sound and visual effects + 1.2.13 + some other things

### PR DESCRIPTION
**Pull request by Misat11**
**Sounds:** ✔️ 
- [x] /playsound command support
- [x] /stopsound command support
- [x] Sound Effects Support
- [x] Break sound effects
- [x] Door, when isn't opened by redstone or button
- [x] Records

**Effects:**
- [x] Splash effect
- [x] Block break effect
- [x] Particle effects (Need some changes because some not working yet) - PC Effect packet
- [x] Particles - PC Particle packet (Working, but some particles aren't in bedrock edition and when you use /particle fallingdust <...>, you got: _DisconnectedEvent io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: Packet "ServerSpawnParticlePacket" not fully read. io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: Packet "ServerSpawnParticlePacket" not fully read._)

**To future:**
- [x] Complete sound list in SoundTranslator (missing maybe 40-50 translates)
- [x] Complete sound list in PCSoundEventPacketTranslator  (need tests and fix some translates)
- [x] Other sound effects (maybe some not working, I don't know what sounds)
- [x] Jukebox popup

This pull request contains fix to teleport #304 and maybe some similar issues (after far distances is problem with blocks and then with chunks, some problems are logged to console (debug) and teleport to blocks is problem and sometimes teleport don't working as previously)
**And this request also has 1.2.13 update https://github.com/DragonetMC/DragonProxy/pull/305/commits/6017e40d220167fb094a765df9a6a7b60a160826 😃**

### CODE REVIEW

**Comment by lukeeey** [comment]
Please move `SoundTranslator` to [here](https://github.com/DragonetMC/DragonProxy/tree/master/proxy/src/main/java/org/dragonet/proxy/network/translator).

File: proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCPlayEffectPacketTranslator.java [hunk 1]
```diff
@@ -156,5 +192,48 @@
 		breakBlock.put(12, BuiltinSound.BLOCK_SAND_BREAK);
 		breakBlock.put(13, BuiltinSound.BLOCK_GRAVEL_BREAK);
 	}
+	
+	// From Nukkit
+    public static final int PARTICLE_TYPE_BUBBLE = 1;
```
Comment (lukeeey) [inline proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCPlayEffectPacketTranslator.java]: Nukkit is disgusting.

Can you make a new enum somewhere when you have time named "ParticleEffect"? Then make the names like "TYPE_CRITICAL" without the PARTICLE prefix.

Maybe in the commons module, somewhere like [here](https://github.com/DragonetMC/DragonProxy/tree/master/commons/src/main/java/org/dragonet/common/data).

Comment (Misat11) [reply proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCPlayEffectPacketTranslator.java]: ok, I do it

**Comment by Misat11** [comment]
I will continue in this pull request after Easter because now I don't have enough time.

**Comment by lukeeey** [comment]
You've already done so much so thanks :)

File: proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCPlaySoundPacketTranslator.java [hunk 2]
```diff
@@ -14,37 +14,47 @@
...
-    }
+	public PEPacket[] translate(UpstreamSession session, ServerPlaySoundPacket packet) {
+		try {
+			String soundName;
+			
+			if (BuiltinSound.class.isAssignableFrom(packet.getSound().getClass())) {
+				if(DragonProxy.getInstance().getSoundTranslator().isIgnored((BuiltinSound)packet.getSound())) {
+					return null;
+				}
+				if (DragonProxy.getInstance().getSoundTranslator().isTranslatable((BuiltinSound)packet.getSound())) {
```
Comment (lukeeey) [inline proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCPlaySoundPacketTranslator.java:37]: Why does everyone, even @HoverEpic, do this?

UpstreamSession has a getProxy() method.

Comment (lukeeey) [reply proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCPlaySoundPacketTranslator.java]: I'll change it myself im just saying for future reference.

**Comment by DefinitlyEvil** [comment]
four spaces = GOOD! 
tabs = SUCKS! 
:/

**Comment by Misat11** [comment]
@DefinitlyEvil ok, when I have time I will change tabs to spaces

**Review by HoverEpic — COMMENTED** [review]
Nice update, thank you I'll look à this tomorrow.

**Comment by HoverEpic** [comment]
@Misat11 can you separate PRs? i can't just merge the 1.2.13 without the WIP sound things.
If you want to PR on multiple subjects, think about this please.

**Comment by Misat11** [comment]
@HoverEpic I don't know that I can separate it, I did more than just the sounds (and 1.2.13) and I don't know if my update (1.2.13) working without it. I will try to translate the remaining sounds today, and for other things (particle effects, block break animation) I can make new pull requests.

**Comment by HoverEpic** [comment]
Ok, so we'll just wait you finish this PR to merge. For next time, try to separate things =P.

**Comment by Misat11** [comment]
Ok, sounds working (some sounds are broken (only sometimes) or play other sounds, this need to test and find wrong translates), for effects I'll create new pull requests and done it (currently I don't know how to complete this translates), 1.2.13 is working but has some new bugs with middle click, but other things are working and you can merge it

**Comment by lukeeey** [comment]
i dont see the point in splitting up PRs.

**Comment by HoverEpic** [comment]
I think a PR per feature is a good thing to keep things organized. Here I speak about sounds and protocol update. Now it's too late for splitting, because without protocol update nothing works.

**Comment by Misat11** [comment]
ok, d27afd1 was my last commit to this PR, you can merge it

**Comment by oldgalileo** [comment]
I've been running this and it works for the most part. Two big things are wearing armor seems to not be properly rendered in the inventory, and enchantments don't render. Besides that, connecting and moving about works great.

**Comment by Misat11** [comment]
@HowardStark I know there are big issues with inventories, but they aren't target of this pull request. But once I will definitely work on this.

**Pull request merged**